.github/workflows/pr-jobs.yml

```diff
@@ -461,19 +462,29 @@ jobs:
     runs-on:
       labels: 4-core-ubuntu
     container:
-      image: zjay437/rocksdb:0.6
+      image: evolvedbinary/rocksjava:centos6_x64-be
       options: --shm-size=16gb
     steps:
-    - uses: actions/checkout@v4.1.0
+    # The docker image is intentionally based on an OS that has an older GLIBC version.
+    # That GLIBC is incompatibile with GitHub's actions/checkout. Thus we implement a manual checkout step.
+    - name: Checkout
+      env:
+        GH_TOKEN: ${{ github.token }}
+      run: |
```

Comment:
This is copy-pasting a complex piece of functionality. Presumably it can go into a sharable "uses" step.

Reply:
@pdillinger I did try that, but unfortunately it is not possible as you end up in a "chicken or egg" situation. i.e. if you put the code in a separate GitHub action file of this repo, then it isn't yet available as the code hasn't been checked out from Git until after this step would run.